interfaces/builtin/docker_support.go

```diff
@@ -594,6 +595,16 @@ var (
 	parserFeatures = release.AppArmorParserFeatures
 )
 
+func (iface *dockerSupportInterface) UDevConnectedPlug(spec *udev.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	var controlsDeviceCgroup bool
+	_ = plug.Attr("controls-device-cgroup", &controlsDeviceCgroup)
+	if controlsDeviceCgroup {
+		spec.SetControlsDeviceCgroup()
+	}
```

Comment:
I do recall discussing this and saying I thought it was fine to use spec.SetControlsDeviceCgroup() in this interface, but I'm a little surprised you aren't just setting `controlsDeviceCgroup: true,` (like we did in greengrass) and instead using an interface attribute. Using an attribute suggests that you expect a mixture of cases for when snaps plugging this interface sometimes want the device cgroup and sometimes don't. Is there ever a time when the cgroup is in effect that snaps would want the device cgroup? It seems like the plugging snap would always die when other interfaces are used, so I'm confused why this needs to be conditional on the snap developer's choice. Perhaps I am missing something?

Put another way, if something plugs docker-support, should we assume that it will manage cgroups and therefore use `controlsDeviceCgroup: true,`. I think we can, but again, I maybe am missing something.

Reply:
Well we have a specific use case with Azure IoT Edge in that it uses docker/moby for container management on specifically IoT devices, and thus needs to manage cgroups for containers (called modules in Azure IoT Edge speak) that interact with devices such as gpio, etc.

However, I don't see why we need to also enable this for the other users of the docker-support interface such as the docker snap itself or microk8s, etc. From my notes I think it was @pedronis who recommended using an attribute for it in Malta. However, at least for the docker snap currently, no interfaces are plugged that trigger udev/device cgroup currently so it probably won't trigger any new issues with the docker snap.

Reply:
> However, at least for the docker snap currently, no interfaces are plugged that trigger udev/device cgroup currently

Right. I think thus far we've been fortunate or the docker snap was written in such a way as to specifically avoid this. My understanding of docker is that use of cgroups is not optional (please correct me) but even it it was, they are always expected to work when configured. I maintain it should be unconditional since:
* this will avoid a potential future bug if the docker snap adds a new interface that triggers them (or more likely, an existing interface that docker plugs starts to trigger them)
* there is no regression in security (and the lack of the attribute is doing nothing for the real-world docker snap)
* docker-support policy is advisory only (it is trivial for something plugging docker-support to own the system, with or without the device cgroup in effect)

I appreciate the desire to limit this to where it is needed, but this interface is the *docker*-support interface and I can't imagine a docker install that would be fully functional without being able to modify cgroups.

Reply:
Okay, I will update this to be unconditional